workflow/controller/operator.go

```diff
@@ -153,6 +157,7 @@ func newWorkflowOperationCtx(wf *wfv1.Workflow, wfc *WorkflowController) *wfOper
 		volumes:                wf.Spec.DeepCopy().Volumes,
 		completedPods:          make(map[string]bool),
 		succeededPods:          make(map[string]bool),
+		podGCSelectedPods:      make(map[string]bool),
```

Comment:
I don't think we should have another map, instead we should filer when we add to the queue, pod should be available in informer

Reply:
Thanks. Removed this unnecessary map.